src/stk/ea/mutation/mutators/molecule/random_smarts.py

```diff
@@ -0,0 +1,121 @@
+"""
+Random Smarts
+=============
+
+"""
+
+import numpy as np
+
+
+from .mutator import MoleculeMutator
+from ...records import MutationRecord
+from ....molecule_records import MoleculeRecord
+from rdkit import Chem
+from .....molecular import BuildingBlock
+
+
+class RandomSmarts(MoleculeMutator):
+    """
+    Substitutes functional groups within building blocks.
+
+    This mutator takes a :class:`ConstructedMolecule` and substitutes
+    the existing building blocks with ones containing
+    new functionalities, as specified in the SMARTS strings.
+    Atoms in functional groups cannot be adjusted as these are used to
+    construct the :class: `ConstructedMolecule`, and will
+    result in an error being raised.
+    Only external functionalities can be replaced.
```

Comment:
Probably worth adding an examples section like in the other mutators

Reply:
Added an example